BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -341,6 +344,25 @@ public final class BlueprintView: UIView {
         setNeedsLayout()
     }
 
+    /// Updates the layout caching flag for `ElementState` only from root level `BlueprintView`s
+    /// For out-of-band renders, this should re-use the flag value used by the parent view.
+    /// This assumes all blueprint layout calls happen on the main thread.
+    private func updateLayoutCachingFlag(_ closure: () -> Void) {
+        print("cross layout caching: \(ElementState.enableCrossLayoutCaching)")
+        var shouldClearCrossLayoutCaching = false
+        if ElementState.enableCrossLayoutCaching == nil {
+            shouldClearCrossLayoutCaching = true
+            ElementState.enableCrossLayoutCaching = isCrossLayoutCachingEnabled
+        }
+
+        defer {
+            if shouldClearCrossLayoutCaching {
+                ElementState.enableCrossLayoutCaching = nil
+            }
+        }
```

Comment:
I might actually move this to `updateViewHierarchyIfNeeded`. Less elegant but safer, we don't want to call `updateViewHierarchyIfNeeded` without this code and we aren't wrapping any other methods with it.